test/e2e/self_hosted.go

```diff
@@ -396,6 +396,10 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 		if selfHostedNamespace != nil {
 			// Dump all Cluster API related resources to artifacts before pivoting back.
 			dumpAllResources(ctx, selfHostedClusterProxy, input.ArtifactFolder, namespace, clusterResources.Cluster)
+
+			// Dump logs from the Machines and infrastructure.
+			input.BootstrapClusterProxy.CollectWorkloadClusterLogs(ctx, namespace.Name, clusterResources.Cluster.Name, filepath.Join(input.ArtifactFolder, "clusters", clusterResources.Cluster.Name))
```

Comment:
Would be better to find a relatively easy way to inject the log collector into the workload cluster proxy, but this should work for now.

Reply:
I'm slightly confused. dumpAllResources already collects logs via the selfHostedClusterProxy for apparently the same cluster.

Is this intentional?

Reply:
selfHostedClusterProxy doesn't have a clusterLogger to collect the logs from the Machine and infrastructure before deletion. I was testing if this approach would collect them, but I've reworked it to pass the logger to the WorkloadClusterProxy.